### DISCUSSION TITLE
Check that the z3 version is the one we expect

### DISCUSSION THREAD

**Discussion by utaal** [verus-lang/verus#235] [General]
If not, we may want to panic.

**Comment by utaal** [comment]
Possibly addressed by https://github.com/secure-foundations/verus/pull/245, which asks z3 and hardcodes the “current expected” version, and panics if it doesn’t match.

**Reply by parno** [reply]
Would it be worth considering a warning instead of a panic?  Or perhaps a command-line switch to ignore the results of the check?  We might want to experiment with other versions of Z3, or have the CI run with multiple versions, for example.

**Reply by tjhance** [reply]
i think the check is a good idea, but as bryan suggests, there ought to be a command line flag to opt out of the check.

**Reply by utaal** [reply]
Makes sense. I'm going to go with the panic with a flag to disable the check, because the warning may not be immediately visible when running tests.

**Comment by utaal** [comment]
This is implemented.
